### PR TITLE
Feature/devsu 1417 add loading bar to write operations

### PR DESCRIPTION
**Pull request by Anna-log7**
- Created a loading spinner button for write operations
- Added the `AsyncButton` in place of normal `Button`s where write operations are done
- Fixed some props/added more types

### CODE REVIEW

**Review by kttkjl — COMMENTED** [review]
code makes sense, just some type errors that needs to be addressed as things get changed

File: app/components/AsyncButton/index.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  children,
+  isLoading,
+  onClick,
+  ...buttonProps
+}: AsyncButtonProps): JSX.Element => {
+  const [loadingStarted, setLoadingStarted] = useState(false);
+
+  useEffect(() => {
+    if (!isLoading && loadingStarted) {
+      window.setTimeout(() => setLoadingStarted(false), 3000);
```
[the remark this answers is not in the document]
Comment (Anna-log7) [reply app/components/AsyncButton/index.tsx:28]: It's an arbitrary 3 seconds to have a smooth animation

File: .eslintrc.json [hunk 2]
```diff
@@ -100,6 +100,7 @@
     "class-methods-use-this": "off",
     "react/jsx-filename-extension": "off",
     "@typescript-eslint/no-empty-function": "off",
-    "react/jsx-indent": "error"
+    "react/jsx-indent": "error",
+    "react/require-default-props": "off"
```
[the remark this answers is not in the document]
Comment (Anna-log7) [reply .eslintrc.json:104]: I'll be fixing them as I run into them for the most part

File: app/views/ReportView/components/Immune/index.tsx [hunk 3]
```diff
@@ -1,13 +1,13 @@
 import React, { useEffect, useState, useContext } from 'react';
 
-import DataTable from '../../../../components/DataTable';
-import Image from '../../../../components/Image';
-import ReportContext from '../../../../components/ReportContext';
-import api, { ApiCallSet } from '../../../../services/api';
-import { hlaColumnDefs, cellTypesColumnDefs } from './columnDefs';
 import DemoDescription from '@/components/DemoDescription';
+import DataTable from '@/components/DataTable';
+import Image from '@/components/Image';
```
[the remark this answers is not in the document]
Comment (Anna-log7) [reply app/views/ReportView/components/Immune/index.tsx:5]: I actually changed this to use the `ImageType` in the `Image` component in one of the other PRs that are open

**Review by kttkjl — APPROVED** [review]

**Pull request merged**